catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/RoleRepository.java

```diff
@@ -45,10 +53,30 @@ public RoleRepository(CollectionDAO dao) {
 
   @Override
   public Role setFields(Role role, Fields fields) throws IOException {
-    // Nothing to set.
+    role.setPolicy(fields.contains("policy") ? getPolicyForRole(role) : null);
     return role;
   }
 
+  private EntityReference getPolicyForRole(Role role) throws IOException {
+    List<String> result =
+        daoCollection
+            .relationshipDAO()
+            .findFrom(
+                role.getId().toString(),
+                Entity.ROLE,
+                Relationship.APPLIED_TO.ordinal(),
```

Comment:
should this be "belongs_to" not sure if we need "applied_to" as another relationship

Reply:
we don't have "belongs_to" more like "contains"? @mithmatt

Reply:
Updated to role -- contains --> policy